packages/serverless/src/domain/use-case/document.ts

```diff
@@ -312,7 +301,7 @@ async function deleteDocument(
   throw Error('Document not found');
 }
 
-async function findDocumentWithMetadata(
+async function listDocumentWithMetadata(
   databaseName: string,
   collectionName: string,
```

Comment:
at line `321` try avoid calling { client } from DATABASE_ENGINE
You can use
```ts
    const { db } = new ModelDatabase();
```

Reply:
it's a join query so I don't think it belongs to any model

Reply:
only ModelDatabase is wrap the `db: Db` of MongoDB, it was born for manage thing related to system database with mongodb, not `ModelMetadataDatabase`

Reply:
fixed